ui/src/main/webapp/src/app/reports/technologies/tech-report.service.ts

```diff
@@ -22,10 +28,109 @@ export class TechReportService extends GraphService
             includeInVertices: false
         });
     }
- }
+
+    getEjbMessageDrivenModel(execID: number, filter?: ReportFilter): Observable<EJBStatDTO[]> {
+        return this.getEJBs<EjbMessageDrivenModel>(execID, 'mdb', EjbMessageDrivenModel, filter);
+    }
+
+    getEjbSessionBeanModel(execID: number, sessionType: string, filter?: ReportFilter): Observable<EJBStatDTO[]> {
+        return this.getEJBs<EjbSessionBeanModel>(execID, 'ejb', EjbSessionBeanModel, filter, sessionType);
+    }
+
+    getEjbEntityBeanModel(execID: number, filter?: ReportFilter): Observable<EJBStatDTO[]> {
+        return this.getEJBs<EjbEntityBeanModel>(execID, 'entity', EjbEntityBeanModel, filter);
+    }
+
+    private getEJBs<T extends EjbBeanBaseModel>(execID: number, ejbType: string, clazz?: typeof EjbBeanBaseModel,  filter?: ReportFilter, sessionType?: string): Observable<EJBStatDTO[]> {
+        let serializedFilter = this.serializeFilter(filter);
+        let url =`${Constants.GRAPH_REST_BASE}/reports/${execID}/ejb/${ejbType}`;
+        if (sessionType) {
+            url += '?sessionType=' + sessionType;
+        }
+        return this._http.post(url, serializedFilter, this.JSON_OPTIONS)
+            .map(res => res.json())
+            .map(data => {
+                if (!Array.isArray(data)) {
+                    throw new Error("No items returned");
+                }
+
+                return <T[]>this._graphJsonToModelService.fromJSONarray(data, clazz);
+            })
+            .map(value => {
+                    return this.loadStats(value);
+                }
+            );
+    }
+
+    private loadStats(values:EjbBeanBaseModel[]):EJBStatDTO[] {
```

Comment:
Are "loadStats" and "EJBStatDTO" really the best names for these? It seems to me that these aren't really statistics entries. Maybe "EJBInformation" or something would be better?